applications/GeoMechanicsApplication/custom_elements/axisymmetric_stress_state_policy.cpp

```diff
@@ -0,0 +1,63 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//                   Marjan Fathian
+//
+
+#include "axisymmetric_stress_state_policy.h"
+#include "custom_utilities/element_utilities.hpp"
+
+namespace Kratos
+{
+
+static const SizeType NumberOfUDofPerNode = 2; // Axi-symmetric stress state assumes a 2D problem definition
+
+Matrix AxisymmetricStressState::CalculateBMatrix(const Matrix&         GradNpT,
+                                                       const Vector&         Np,
+                                                       const Geometry<Node>& rGeometry) const
+{
+    const double radius = GeoElementUtilities::CalculateRadius(Np, rGeometry);
+
+    const auto dimension       = rGeometry.WorkingSpaceDimension();
+    const auto number_of_nodes = rGeometry.size();
+    Matrix result = ZeroMatrix(VOIGT_SIZE_2D_AXISYMMETRIC, NumberOfUDofPerNode * number_of_nodes);
```

Comment:
That makes sense for these continuum elements. The number of U D.O.F. equals the dimension. It will be different for structural elements like beams and shells and for Cosserat type continuum elements.